src/main/groovy/eu/spaziodati/azkaban/jobtype/GroovyRemoteJob.groovy

```diff
@@ -105,6 +106,18 @@ class GroovyRemoteJob extends GroovyProcessJob {
         return jarpath.toString()
     }
 
+    // Make sure that remote dir is unique
+    def getRemoteDir(jobProps) {
+        def dir = jobProps.getString(REMOTE_DIR, null)
+        if (dir == null) {
+            def tempDir = File.createTempFile('azkaban-', '-default-dir')
```

Comment:
This is not fine. Recall that this code is executed on the azkaban server: you are creating a temp directory on the azkaban machine, that will never be used. Additionally even if you call `deleteOnExit()` this refers to the exit of azkaban server process, that usually does not occur very frequently... If you run a lot of jobs you will fill the `/tmp` folder with a lot of unused and empty subfolders
At the end this works, because the random name used for that unused temporary directory is used to create a directory on the remote machine (if the remote directory  does not exist, the plugin will call `mkdirs`)
So I think it would be better to simply generate a random name. Just append a random sequence (this should be ok http://stackoverflow.com/a/41156 )
imho it would be better to keep the reference to the exec-id in the name, something like `/tmp/azkaban-${flowid}-${randomstring}`